src/data/set/intervals/basic.lean

```diff
@@ -84,6 +84,15 @@ lemma Ioi_def (a : α) : {x | a < x} = Ioi a := rfl
 @[simp] lemma mem_Ici : x ∈ Ici a ↔ a ≤ x := iff.rfl
 @[simp] lemma mem_Ioi : x ∈ Ioi a ↔ a < x := iff.rfl
 
+instance decidable_mem_Ioo : Π [decidable (a < x ∧ x < b)], decidable (x ∈ Ioo a b) := id
```

Comment:
AFAIR (not sure), Lean sometimes fails to unfold `id` while matching different instances. So, I suggest
```suggestion
instance decidable_mem_Ioo [h : decidable (a < x ∧ x < b)] : decidable (x ∈ Ioo a b) := h
```

Reply:
Here is an example (fails):
```lean
example (α : Type*) (x : α) : id x = x := by tactic.reflexivity tactic.transparency.instances
```

Reply:
Not here it isn't; try it and see. `‹decidable _›` works, but that's not shorter than `by assumption`.